build.sh

```diff
@@ -43,29 +43,31 @@ COMPILEFLAGS="-DCMAKE_INSTALL_PREFIX="
 export DESTDIR="$(cd "$(dirname "$0")" && pwd)"
 BUILDTYPE="$(color 6 release)"
 SCRIPT_ARGC=1 # number of arguments eaten by this script
-if [ "$ARG_LENGTH" -gt 0 -a "$1" == "--debug" -o "$2" == "--debug" ]; then
+if [ "$ARG_LENGTH" -gt 0 -a "$1" = "--debug" -o "$2" = "--debug" ]; then
   COMPILEDIR="debug_build"
   COMPILEFLAGS="$COMPILEFLAGS -DCMAKE_BUILD_TYPE=DEBUG"
   BUILDTYPE="$(color 1 debug)"
   SCRIPT_ARGC=$((SCRIPT_ARGC+1))
 fi
-if [ "$ARG_LENGTH" -gt 0 -a "$1" == "--recompile" -o "$2" == "--recompile" ]; then
+if [ "$ARG_LENGTH" -gt 0 -a "$1" = "--recompile" -o "$2" = "--recompile" ]; then
   STRCOMPILE="$(color 5 Recompiling)"
   rm -rf $COMPILEDIR
   SCRIPT_ARGC=$((SCRIPT_ARGC+1))
 fi
-if [ "$CROSS" == "mingw64" ]; then
+if [ "$CROSS" = "mingw64" ]; then
   COMPILEFLAGS="$COMPILEFLAGS -DCMAKE_TOOLCHAIN_FILE=../Toolchain-mingw64.cmake"
 fi
-COMPILEFLAGS="$COMPILEFLAGS ${@:${SCRIPT_ARGC}}" # pass other arguments to CMAKE
+SUBSTRING=$(echo $@ | cut -c${SCRIPT_ARGC}-)
```

Comment:
Isn't it supposed to be `cut -f`? Not sure.

Reply:
I was confused about that, too. I think the original implementation intended for it to separate by argument, not character, but bash's `${parameter:offset}` syntax is character indexed. Here is the documentation: https://www.gnu.org/software/bash/manual/bashref.html#Shell-Parameter-Expansion

Now is a reasonable time to change it to `-f` if we want to correct the behavior. Want me to drop a commit on top or fix it up?

Reply:
This documentation explains that the behaviour is different for `$@`. This is this specific behaviour that I expected when I wrote this piece of code, which would match with `cut -f` rather than `cut -c`.
However, command line parameters with spaces enclosed in `""` are treated as a single parameter with `$@`, but with `cut -f` they would be separated. In our case it's not an issue, but that's an important thing to remember.